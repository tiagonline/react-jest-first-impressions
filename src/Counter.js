import React, { Component } from 'react';

class Counter extends Component {
  render() {
    return (
      <div className="Counter">
        <h1>VocÃª tem { this.props.count * 2} counters.</h1>
      </div>
    );
  }
}

export default Counter;
