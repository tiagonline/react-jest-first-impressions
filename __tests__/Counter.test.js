import React from 'React';
import { shallow } from 'enzyme';
import Counter from '../src/Counter';

describe('Testing Counter Component', () => {
    test('should render correctly', () => {
        const wrapper = shallow(
            <Counter count={5} />
        );
        
        expect(wrapper).toMatchSnapshot();
        expect.setProps({ count:10 });
        expect(wrapper).toMatchSnapshot();        
    });
});