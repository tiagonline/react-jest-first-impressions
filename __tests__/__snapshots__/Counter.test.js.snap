// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    count={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Você tem 
        10
         counters.
</h1>,
      "className": "Counter",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Você tem ",
          10,
          " counters.",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Você tem ",
        10,
        " counters.",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Você tem 
          10
           counters.
</h1>,
        "className": "Counter",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Você tem ",
            10,
            " counters.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Você tem ",
          10,
          " counters.",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
